internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	RefreshTokenLength                          types.Int64  `tfsdk:"refresh_token_length"`
+	RollRefreshTokenValues                      types.Bool   `tfsdk:"roll_refresh_token_values"`
+	RefreshTokenRollingGracePeriod              types.Int64  `tfsdk:"refresh_token_rolling_grace_period"`
+	RefreshRollingInterval                      types.Int64  `tfsdk:"refresh_rolling_interval"`
+	PersistentGrantReuseGrantTypes              types.Set    `tfsdk:"persistent_grant_reuse_grant_types"`
+	PersistentGrantContract                     types.Object `tfsdk:"persistent_grant_contract"`
+	BypassAuthorizationForApprovedGrants        types.Bool   `tfsdk:"bypass_authorization_for_approved_grants"`
+	AllowUnidentifiedClientROCreds              types.Bool   `tfsdk:"allow_unidentified_client_ro_creds"`
+	AllowUnidentifiedClientExtensionGrants      types.Bool   `tfsdk:"allow_unidentified_client_extension_grants"`
+	AdminWebServicePcvRef                       types.Object `tfsdk:"admin_web_service_pcv_ref"`
+	AtmIdForOAuthGrantManagement                types.String `tfsdk:"atm_id_for_oauth_grant_management"`
+	ScopeForOAuthGrantManagement                types.String `tfsdk:"scope_for_oauth_grant_management"`
+	AllowedOrigins                              types.Set    `tfsdk:"allowed_origins"`
+	UserAuthorizationUrl                        types.String `tfsdk:"user_authorization_url"`
+	BypassActivationCodeConfirmation            types.Bool   `tfsdk:"bypass_activation_code_confirmation"`
+	RegisteredAuthorizationPath                 types.String `tfsdk:"registered_authorization_path"`
+	PendingAuthorizationTimeout                 types.Int64  `tfsdk:"pending_authorization_timeout"`
+	DevicePollingInterval                       types.Int64  `tfsdk:"device_polling_interval"`
+	ActivationCodeCheckMode                     types.String `tfsdk:"activation_code_check_mode"`
+	UserAuthorizationConsentPageSetting         types.String `tfsdk:"user_authorization_consent_page_setting"`
+	UserAuthorizationConsentAdapter             types.String `tfsdk:"user_authorization_consent_adapter"`
+	ApprovedScopesAttribute                     types.String `tfsdk:"approved_scopes_attribute"`
+	ApprovedAuthorizationDetailAttribute        types.String `tfsdk:"approved_authorization_detail_attribute"`
+	ParReferenceTimeout                         types.Int64  `tfsdk:"par_reference_timeout"`
+	ParReferenceLength                          types.Int64  `tfsdk:"par_reference_length"`
+	ParStatus                                   types.String `tfsdk:"par_status"`
+	ClientSecretRetentionPeriod                 types.Int64  `tfsdk:"client_secret_retention_period"`
+	JwtSecuredAuthorizationResponseModeLifetime types.Int64  `tfsdk:"jwt_secured_authorization_response_mode_lifetime"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *oauthAuthServerSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```

Comment:
We don't need to split this into separate methods like the PD provider does if we don't have default_ versions of resources. The same applies to the other methods (read, update, etc.)